pygeos/test/test_set_operations.py

```diff
@@ -105,6 +143,92 @@ def test_set_operation_reduce_all_none_arr(n, func, related_func):
+def test_set_operation_prec_reduce_grid_size_nan(func, related_func):
+    actual = func([point, point], grid_size=np.nan)
+    assert actual is None
+
+
+@pytest.mark.skipif(pygeos.geos_version < (3, 9, 0), reason="GEOS < 3.9")
+@pytest.mark.parametrize("n", range(1, 5))
+@pytest.mark.parametrize("func, related_func", REDUCE_SET_OPERATIONS_PREC)
+@pytest.mark.parametrize("grid_size", [0, 1])
+def test_set_operation_prec_reduce_1dim(n, func, related_func, grid_size):
+    actual = func(reduce_test_data[:n], grid_size=grid_size)
+    # perform the reduction in a python loop and compare
+    expected = reduce_test_data[0]
+    for i in range(1, n):
+        expected = related_func(expected, reduce_test_data[i], grid_size=grid_size)
+
+    assert pygeos.equals(actual, expected)
+
+
+@pytest.mark.skipif(pygeos.geos_version < (3, 9, 0), reason="GEOS < 3.9")
+@pytest.mark.parametrize("func, related_func", REDUCE_SET_OPERATIONS_PREC)
+def test_set_operation_prec_reduce_axis(func, related_func):
+    data = [[point] * 2] * 3  # shape = (3, 2)
+    actual = func(data, grid_size=1)
+    assert actual.shape == (2,)
+    actual = func(data, grid_size=1, axis=0)  # default
+    assert actual.shape == (2,)
+    actual = func(data, grid_size=1, axis=1)
+    assert actual.shape == (3,)
+    actual = func(data, grid_size=1, axis=-1)
+    assert actual.shape == (3,)
+
+
+@pytest.mark.skipif(pygeos.geos_version < (3, 9, 0), reason="GEOS < 3.9")
+@pytest.mark.parametrize("none_position", range(3))
+@pytest.mark.parametrize("func, related_func", REDUCE_SET_OPERATIONS_PREC)
+def test_set_operation_prec_reduce_one_none(func, related_func, none_position):
+    test_data = reduce_test_data[:2]
+    test_data.insert(none_position, None)
+    actual = func(test_data, grid_size=1)
+    expected = related_func(reduce_test_data[0], reduce_test_data[1], grid_size=1)
+    assert pygeos.equals(actual, expected)
+
+
+@pytest.mark.skipif(pygeos.geos_version < (3, 9, 0), reason="GEOS < 3.9")
+@pytest.mark.parametrize("none_position", range(3))
+@pytest.mark.parametrize("func, related_func", REDUCE_SET_OPERATIONS_PREC)
+def test_set_operation_prec_reduce_two_none(func, related_func, none_position):
+    test_data = reduce_test_data[:2]
+    test_data.insert(none_position, None)
+    test_data.insert(none_position, None)
+    actual = func(test_data, grid_size=1)
+    expected = related_func(reduce_test_data[0], reduce_test_data[1], grid_size=1)
+    assert pygeos.equals(actual, expected)
+
+
+@pytest.mark.skipif(pygeos.geos_version < (3, 9, 0), reason="GEOS < 3.9")
+@pytest.mark.parametrize("n", range(1, 3))
+@pytest.mark.parametrize("func, related_func", REDUCE_SET_OPERATIONS_PREC)
+def test_set_operation_prec_reduce_all_none(n, func, related_func):
+    # API change: before, union_all([None]) yielded EMPTY GEOMETRYCOLLECTION
+    # The new behaviour is that it returns None if all inputs are None.
+    assert func([None] * n, grid_size=1) is None
```

Comment:
Are those None-handling tests strictly needed? Because that's implemented on the python side and is the same regardless of passing grid_size or not? (now, implementation can change in the future, of course)

Reply:
It is implemented at the C level instead of Python level, and since `grid_size` goes to a different C function, I wanted to verify that all the existing behavior was consistent.